connector/src/main/java/org/jboss/as/connector/subsystems/datasources/GetDataSourceClassInfoOperationHandler.java

```diff
@@ -0,0 +1,235 @@
+        if (context.isNormalServer()) {
+            context.addStep(new OperationStepHandler() {
+
+                @Override
+                public void execute(final OperationContext context, final ModelNode operation) throws OperationFailedException {
+
+                    ServiceController<?> sc = context.getServiceRegistry(false).getRequiredService(
+                            ConnectorServices.JDBC_DRIVER_REGISTRY_SERVICE);
+                    DriverRegistry driverRegistry = DriverRegistry.class.cast(sc.getValue());
+                    InstalledDriver driver = driverRegistry.getInstalledDriver(driverName);
+                    if (driver == null) {
+                        context.getResult().set(new ModelNode());
+                        return;
+                    }
+                    ModelNode result = dsClsInfoNode(driver.getModuleName(), driver.getDataSourceClassName(),
+                            driver.getXaDataSourceClassName());
+                    context.getResult().set(result);
+                }
+
+            }, OperationContext.Stage.RUNTIME);
+        }
+    }
+
+    static ModelNode dsClsInfoNode(ModuleIdentifier mid, String dsClsName, String xaDSClsName) throws OperationFailedException {
+        ModelNode result = new ModelNode();
+        if (dsClsName != null) {
+            ModelNode dsNode = new ModelNode();
+            dsNode.get(dsClsName).set(findPropsFromCls(mid, dsClsName));
+            result.add(dsNode);
+        }
+        if (xaDSClsName != null) {
+            ModelNode xaDSNode = new ModelNode();
+            xaDSNode.get(xaDSClsName).set(findPropsFromCls(mid, xaDSClsName));
+            result.add(xaDSNode);
+        }
+        return result;
+    }
+
+    private static ModelNode findPropsFromCls(ModuleIdentifier mid, String clsName) throws OperationFailedException {
+        final ModelNode result = new ModelNode();
+        findPropsMap(mid, clsName).forEach((k, v) -> {
+            result.get(k).set(v.getTypeName());
+        });
+        return result;
+    }
+
+    private static Map<String, Type> findPropsMap(ModuleIdentifier mid, String clsName) throws OperationFailedException {
+        Class<?> cls = null;
+        ClassLoader cl = null;
+        if (mid != null) {
+            try {
+                cl = Module.getCallerModuleLoader().loadModule(mid).getClassLoader();
+            } catch (ModuleLoadException e) {
+                throw new OperationFailedException(ConnectorLogger.SUBSYSTEM_DATASOURCES_LOGGER.failedToLoadModuleDriver(mid
+                        .toString()), e);
+            }
+        }
+        if (cl == null) {
+            cl = getClassLoader(GetDataSourceClassInfoOperationHandler.class);
+        }
+        try {
+            cls = Class.forName(clsName, true, cl);
+        } catch (ClassNotFoundException e) {
+        }
+        if (cls == null) {
+            throw ConnectorLogger.SUBSYSTEM_DATASOURCES_LOGGER.failedToLoadDataSourceClass(clsName);
```

Comment:
Why not just throw this from the ClassNotFoundException block, that was you can include the original exception which may have additional information